DOCS-10770-X12-2.6-ESB (#1442)
mulesoft/docs-connectors: x12-edi/2.6/modules/ROOT/pages/x12-edi-connector-reference.adoc

= X12 EDI Connector 2.6 - Reference - Mule 4
:page-aliases: connectors::x12-edi/x12-edi-connector-reference.adoc

== Default Configuration

=== Parameters

[%header%autowidth.spread]
|===
| Name | Type | Description | Default Value | Required
|Name | String | The name for this configuration. Connectors reference the configuration with this name. | |x

> | Expiration Policy a| ExpirationPolicy |  Configures the minimum amount of time that a dynamic configuration instance can remain idle before the runtime considers it eligible for expiration. This does not mean that the platform expires the instance at the exact moment that it becomes eligible. The runtime actually purges the instances as it sees fit. |  |

Reviewer: ```suggestion
| Expiration Policy a| ExpirationPolicy |  Configures the minimum amount of time that a dynamic configuration instance can remain idle before Mule considers it eligible for expiration. |  |
```

| Partner application code a| String |  Application Code for partner in group (GS segment). If this value is set, it affects both send and receive message handling. On the receive side, the GS Application Sender's Code must match this value. On the send side, this value is used as the default GS Application Receiver's Code when not set in the message data. |  |
| Form and validation a| Enumeration, one of:

** X12
** HIPAA_SNIP1
** HIPAA_SNIP2 | Schema form and validation level. Schemas used in the configuration must match the form selected, either X12 or HIPAA. SNIP Type 1 and Type 2 validation only applies to HIPAA schemas. |  X12 |
| Schema definitions a| Array of String | List of paths for schemas to be used by connector. The paths may be either for file system or classpath. |  |x
| Character encoding a| Enumeration, one of:

** PLATFORM
** ASCII
** ISO8859_1
** IBM1047
** UTF8 | Character encoding for messages. This character encoding is always used for both send and receive messages. |  PLATFORM |
| Character set a| Enumeration, one of:

** BASIC
** EXTENDED

> ** UNRESTRICTED | Characters allowed in string data. Invalid characters are replaced by the substitution character, if used, or rejected as errors if no substitution character is set or enabled for receive messages in parser options. Either way the invalid characters are logged, and for receive messages are reported in 997 functional acknowledgments. |  EXTENDED |

Reviewer: ```suggestion
** UNRESTRICTED | Characters allowed in string data. Invalid characters are replaced by the substitution character, if used, or rejected as errors if no substitution character is set or enabled for receive messages in parser options. Either way the invalid characters are logged, and receive messages are reported in 997 functional acknowledgments. |  EXTENDED |
```

Reviewer: ?
